### ISSUE TITLE
ENOENT isn't always meaningful

### ISSUE THREAD

**Issue by cperciva** [Tarsnap/tarsnap#86]
This is not particularly useful:

    tarsnap -cf foo foo
    tarsnap: foo/bar/baz: Cannot stat: No such file or directory
    tarsnap: Error exit delayed from previous errors.

If we're trying to archive a file because we found it in a directory we recursed through, but find that it doesn't exist, then it just means that we lost a race condition.  That's fine; just like "file has grown while being archived" it's a normal occurrence when we're creating backups from a running system, and not something we should treat as an error.

If we were running `tarsnap -cf foo foo/bar/baz` and found that `foo/bar/baz` didn't exist, then it should be treated as an error, of course.

**Comment by cperciva** [comment]
I'm marking this as postponed because I don't think it makes sense to work on this until we move to a newer libarchive and use the new tree-traversal code.  There will need to be some logic for recognizing the difference between "this is a file the user asked us to back up" and "this is something we found during the filesystem traversal", and that's going to depend on the new traversal code.

**Comment by mhoran** [comment]
I had a backup run fail last night because of this. Surprisingly this was the first time I've seen this failure over many years of using Tarsnap. While this still doesn't seem urgent, is there any news here?

**Comment by cperciva** [comment]
@mhoran What exactly do you mean by "had a backup run fail"?  This triggers a warning and non-zero exit from tarsnap but it isn't a fatal error, i.e. it won't stop tarsnap from creating an archive.

... unless something else broke as well?

**Comment by mhoran** [comment]
Ah, OK! I saw the message and presumed that it was an error (it looked nearly exactly like your example.) Seeing that there was an "error" I figured the backup script had bailed but didn't capture the exit code. It does look like the backup completed (based on archive size) so I guess this is fine, just misleading. Sorry for the noise!

**Comment by gperciva** [comment]
Hmm,
```
tarsnap: Error exit delayed from previous errors.
```
and a non-zero exit code looks awfully like a fatal error.

Maybe it would help if we added some kind of
```
tarsnap: foo/bar/baz: Cannot stat: No such file or directory
tarsnap: This might be suspicious, but is probably just a normal consequence of backing up a running system.  Reporting an error in case you want to investigate.
tarsnap: Error exit delayed from previous errors.
```

(That's probably too wordy, but something along those lines could avoid some unnecessary angst.  And since we're writing backup software for the truly paranoid, we should expect users to angst easily!)

**Comment by mhoran** [comment]
> and a non-zero exit code looks awfully like a fatal error.

I can confirm that tarsnap did exit non-zero as my script first creates an archive and then deletes archives > 7 days old. The script bailed before deleting old archives, so tarsnap must have exited non-zero. Ideally in this scenario that wouldn't happen. I do only want to delete old archives if a new archive succeeded, but in this case it succeeded but indicated failure due to the race condition.
